Feature/qOperations

m053m716 opened the pull request:
References #29 

I made some progress on this, but I'm stuck: for some reason it is hanging on `doUnitFilter` and `doLFPExtraction` (without any errors). The progress bar won't update and it does not generate any new files when they are run.

Nabarb committed:
TDT update 2

TDT extraction is now working.
Only for streams

Nabarb committed:
Merge remote-tracking branch 'origin/KUMC' into IIT

Nabarb committed:
Fixed time saving

Now Time is added to the Time field in block

Nabarb committed:
fixed normalized coordinates in takeNotes

Nabarb committed:
maybe fixed takeNotes naming issue

Nabarb committed:
rhs2block bugfix

Nabarb committed:
Block init bugfix

Nabarb committed:
efficiency fixes rhs2block

Nabarb committed:
Fix Tank/Animal names

Changed tank and animal variable names to be consistent with the rest. Creating new tanks and animals now works

Nabarb committed:
Another fix in tank naming convention. 

Plus fixed strange error in parsing datetimes

m053m716 committed:
Merge branch 'KUMC' into IIT

m053m716 committed:
Add code to provide graphical interface for setting a channel-wise mask as to what is included / not included in subsequent analyses. Add code to parse unique probe ID (since for Intan, you could have multiple headstages attached to the same port number).
Next step is to match identical probe/channel numbers for the Sort interface so that they can then be sorted across different recordings, etc. -MM

m053m716 committed:
Add initialization code for Sort object. Fix error in path generation for Block initialization, wherein findCorrectPath.m was assigning the Filt file ID to FiltCAR files, which was causing backwards-compatibility issues. -MM

m053m716 committed:
Fix bug with getSpikeTrain, where order of class and spikeObj.Channels.Sorted.class where switched for ismember function.
Update getStatus to use blockObj.Fields property instead of blockObj.updateStatus method (I think they are redundant but blockObj.Fields is more intuitive to me, unless there is a reason for returning it from a method)
Update getStatus to use strncmpi for the first 3 characters only, so that it is easier to reference (for example so it still recognizes 'Clusters' vs 'clusters' vs 'clustered' vs 'cluster')
Add comments to blockObj.list()
Add parsing for 'Clusters' and 'Sorted' field paths to findCorrectPaths.m. -MM

m053m716 committed:
Remove 'class' overloaded method from DiskData (could create issues with previously-sorted spikes; alternative is to keep it and write a batch script to rename scored files to have 'clu' or some other name instead of 'class', which was a bad variable name choice, my bad).
Almost finished with "init" methods for nigeLab.Sort. -MM

m053m716 committed:
Add getTag method to nigeLab.Block class. Add getAllSpikeData method to nigeLab.Sort class. Upgrade nigeLab.Block methods for accessing spike times, waveforms, and classes.

m053m716 committed:
Fix bug in recursion that wouldn't allow you to set multiple channels by default. -MM

m053m716 committed:
Improve backwards-compatibility for parsing tags in case where they were saved as a giant cell array of strings (from old scoring interface). Add tag retrieval to nigeLab.Sort method getAllSpikeData().

m053m716 committed:
Update spike detection to reflect current version of CPLTools (mostly just stuff with saving correct variables in pars struct, etc.) -MM

Nabarb committed:
Merge remote-tracking branch 'origin/KUMC' into IIT

Nabarb committed:
Usability updates and small fixes

-ReReference now uses diskdata class.
-rhd2Block now is more efficient during time vector extraction and actually saves the time vector.
-rhs2Block saves amp_settle charge_recovery compliance_limit using diskdata.
fixed an extraction bug (related to data organization).
Saves time vector.
Tank now is coherent with the other classes on naming variable

m053m716 committed:
Channel selection figure and spike plot figure with spikes integrated from across multiple Blocks is now working. Hopefully finishing up tomorrow -MM

Nabarb committed:
fixes

Bugfixes from some other naming consistency
do methods form Animal and Tank are now working

Nabarb committed:
DiskData class issue

Added exception to subsref method to handle class method. If called with dot referencing returns the class filed, if called as a function executes the method

Nabarb committed:
TDT update, introducing Events!

- Now tank and animal constructors work with TDT as well! 
- Fixed a bug in ReadTDTheader where an integer overflow was causing wrong file length readings (inherited from the official TDT matlab toolboxes, thanks guys)
 - block calss now has an Events field: it is a cell array where different events structs will be held.  I opted for this solution because each struct can have different fields in this way, but it's definitely not written in stone

m053m716 committed:
Merge branch 'IIT' into KUMC

# Conflicts:
#	+nigeLab/@Block/Block.m
#	+nigeLab/@Block/linkToData.m

m053m716 committed:
Add "Event" template and generic "getEventData" method to retrieve event data from different types of Events (needed this for working on Sort GUI). @nabarb, you might look at the nigeLab.defaults.Event() file and nigeLab.Block.getEventData() to see if that is similar to what you had in mind. -MM

m053m716 committed:
Move "channel matching" for multi-block sets to a method of Block, and incorporate it into "getEventData" method. -MM

m053m716 committed:
Add ability to screen for subset of events by other event field values. -MM

m053m716 committed:
Bugfixes, getEventData() method should be working properly now (tested on single and multi-block arrays for 'value' and 'ts' of 'Sorted'). -MM

m053m716 committed:
Updates to linkToData() in progress. Re-worked the way the Block() template is set up so that it's a little easier to assign things from the outset without having to make a lot of exception handling (hopefully). Not all updates are complete. -MM

m053m716 committed:
Change Block defaults. -MM

m053m716 committed:
Probably will require some re-tooling on the Tank and Animal level (haven't looked at those yet). But this does the RawData extraction successfully and I'm moving on to the modified extracted SD files now. -MM

m053m716 committed:
Add fixes to doUnitFilter and doReReference. Importantly, add compatibility with blockObj.Mask property for blockObj.doReReference; this is critical for making sure that only good channels go into the CAR. -MM

m053m716 committed:
Add the inspiration for nigeLab.

m053m716 committed:
Change hard-coded diskData type for doReReference, doSD, and doUnitFilter. -MM

m053m716 committed:
Add parsing of string channel and numeric channel ID so that it is associated with each element of Channels property struct (in addition to 'probe' sub-field). -MM

m053m716 committed:
Change Block defaults in FileTypes in preparation for new handling in DiskData class for 'Events' FileType. This will be a numeric array (most efficient type in Matlab), for which the first 4 columns correspond to [type, value, tag, ts], and the last n columns correspond to 'snippet'. -MM

m053m716 committed:
Add 'Events' type_ to DiskData. Add properties with Private SetAccess but Public GetAccess so that now the tab complete works for DiskData. -MM

m053m716 committed:
Remove Event "Type" and "TypeID" parameters. I think it's most efficient just to always store the variables as a numeric array (matrix) since that is how Matlab is most efficient. -MM

m053m716 committed:
Block.doSD method should now be working. The previous 'Spikes' files have been split into three filetypes: 'Spikes', 'SpikeFeatures', and 'Artifact'. Each of those follows the 'Event' data structure, per DiskData. -MM

m053m716 committed:
Add handling for logical indexing in DiskData. For events, it is now possible to index via logical statements [e.g. spikes = blockObj.Channels(iCh).Spikes.snippets(blockObj.Channels(iCh).Spikes.value == k)] -MM

m053m716 committed:
Update methods for interacting with Event DiskData. -MM

m053m716 committed:
Add initialization of Sorted and Clusters files to 'get' methods for retrieving related Events data. These are added to doSD method for initializing the Sorted and Clusters for downstream interaction with Sorting interface. -MM

m053m716 committed:
Add method Block.getFileType, and associate it with the appropriate functions. -MM

m053m716 committed:
Add Animal and Tank files to nigeLab.defaults; add updateParams method for each class type to reflect updateParams method for Block (although Pars field is different for Animal and Tank).

m053m716 committed:
Add more parsing for backwards compatibility. -MM

Nabarb committed:
Small patches mainly in variable names

Nabarb committed:
Small naming pathces

Nabarb committed:
Logic update in parameters check and recursion patch


I slightly changed the if else tree to hopefully fix some usage cases., also i re-implemented the recursion logic

m053m716 committed:
Add nigeLogo.

Nabarb committed:
Patches and SD optimization

Animal.m
Animal/init
Block
Block/genPaths.m
Block/genPaths.m
Animal/genPaths.m
Animal/getSaveLocation.m
Block/getFolderTree.m (I scorporated here some code from getSaveLocation)
-------------------> Fixed some bugs with the folder tree crreation. 
tdt2Block.m 
parseNamingMetadata.m 
------------> Fixed variable names

Nabarb committed:
Added cprintf for message formatting

utils/cprintf 
--------------> added. useful to format CL messages
DiskData 
------------> small bugfix

Nabarb committed:
SD performance update

libs/peakseek
---------------> Added. faster than matlab's findpeaks
SNEOThreshold
-----------------> implemented peakseak. Matlab's conv used instead of fastsmooth/default. 10x faster(more or less)
defaults/SD
WaveFeatures
------------------>  when using wavedec, passing filter coefficients as input (pars.LoD, pars.HiD) is usually faster than passing wavelet name. pars defined in defaults.

Nabarb committed:
Merge branch 'KUMC' into dev_IIT

m053m716 committed:
Fix bugs for Events with DiskData. -MM

m053m716 committed:
Moving the clusters around is working - hardest part should be done now! -MM

m053m716 committed:
ctrl+z is working (tentatively)

m053m716 committed:
Fix issue where it was not updating the number properly in the plot name. -MM

m053m716 committed:
Saving is not working, but undo works and confirming channel changes works. I think maybe need to add a method either to DiskData or Block to set class and save it properly. -MM

Nabarb committed:
plotWaves bugfix (Status)

Nabarb committed:
Merge branch 'KUMC' into dev_IIT

Nabarb committed:
Small bugfixes

doSD fixed a bug in the status update, now updates all the channels' status

SpikeImage now loads the default colormap correctly

Nabarb committed:
Features plot update

Added a feature panel. Plots 2d and 3d representation of the feature space. Features can be selected from a dropdown menu.
Barely working condition. TODO:
axes autoscaling,
interactive axes
labels and ticks
circles and cyliners
defaults values

m053m716 committed:
Add listeners to features plots

Added a feature panel. Plots 2d and 3d representation of the feature space. Features can be selected from a dropdown menu.
Barely working condition. TODO:
axes autoscaling - complete
interactive axes - STILL NEEDED (best bet is to make a mesh; see: getframe, and the buttondownfcn methods for SpikeImage class)
labels and ticks - complete
circles and cyliners - x (not that helpful anyways)
defaults values - complete (more or less; just not in the +defaults folder, but maybe its okay to keep these few parameters just in the private properties of the class?)

m053m716 committed:
Fix bug in which state of cluster visibility was not associated to FeaturesUI property (VisibleClusters) -MM

m053m716 committed:
Fix bug in which scatter points were not properly initialized so they couldn't have the visibility turned "on" or "off" leading to errors. -MM

m053m716 committed:
Add cluster-cutting to 2D feature axes. I don't think it's necessary to add to 3D axes as it is not intuitive to do it in 3D space anyways.

Still needed
- Actually working SAVE functionality
- Add zoom with mouse wheel to the features plots as with the spike plots
- Cluster quality metrics (should just display as a text object above the 2D plot)
- Auto-cluster functionality
- More-intuitive UNDO

m053m716 committed:
Save works. (Bug was assigning output to Channels(iCh).SORT instead of Channels(iCh).SORTED ...)

m053m716 committed:
Important update: added PROGRESSCAT class. -MM

m053m716 committed:
Fix bug with finding ProgressCat image file location. Add parameterization for ProgressCat so that he hopefully doesn't get too annoying after a long time (made things to make it speed up the display etc. etc)...

Nabarb committed:
bugfixes and cosmetic code changes

Some fixes here and there and small cosmetic changes in code mainly to have matlab stop complaining about stuff.

For some reason private functions InPolygon did not work for me, now everything is in the utils folder.

FeaturesUI: imfreehand is not recommended anymore, I tried to switch to drawfreehand. lokks like is working fine, we might consider changing also the one in SpikeImg

Block:
added SaveLoc.dir  in Paths to be consistent with the other fields. Changed the save function to reflect changes

Nabarb committed:
Silhuette (cluster quality) update

Release notes:
This update introduces silhuette graphics in the features panel. This is a cluster quality metric that evaluates similarity between the points in the cluster and the cluster itself. Different distances metrics are implemented (at this stage the builtin ones) and selectable from a list (by double clicking on them). We can experiment a little and remove some of them in the future and add some new metrics/quality indexes. I spent a good chunk of yesterday trying to figure out how to use mutual information for this purpose. 

This metrics was chosen because it's local, that means that can give a measure for each cluster. I was thinking about introducing a second barplot on the right of this one (and a second list) for global measures, which give a shingle index to the whole clustering process. There is bazillion of them. WIP

Nabarb committed:
Small functionality update

q button press in SpikeImg now reopens the FeatureUI(if closed).

Added the possibility to rotate the 3d feature plot bu pushing the rotate button. On my implementation for some weird reason this breaks the right click functionality on the 2D plot. To restore it, disable the rotation by clicking the button again.

Nabarb committed:
intan Stim data is now handled as Events!

Nabarb committed:
Optimized filtering procedure

defualts.Filt - you can now chose the filter type here. 
Support for iir, sos and fir filters. Three different filter are already implemented: 4th order butter and elliptic, intan unipolar and a more complex SOS elliptic one. 
doUnitFilter - a more optimized zerophase filtering procedure is implemented. Based on a compiled mex file, thers is still room for improvement in the c code. More or less 60% speedup and identical output as filtiflt. Filtering is done directly in single precision. This can be changed without much of loss in computational performance if needed in future.

Nabarb committed:
Optimized filtering procedure

defualts.Filt - you can now chose the filter type here. 
Support for iir, sos and fir filters. Three different filter are already implemented: 4th order butter and elliptic, intan unipolar and a more complex SOS elliptic one. 
doUnitFilter - a more optimized zerophase filtering procedure is implemented. Based on a compiled mex file, thers is still room for improvement in the c code. More or less 60% speedup and identical output as filtiflt. Filtering is done directly in single precision. This can be changed without much of loss in computational performance if needed in future.

StefanoBuccelli committed:
updatePaths feature

Now it is possible to update the whole folder tree in the structure hierarchy. This is useful when something changes or you have to move the tank folder to a new location.

defaults.Block added the field FileNames. Fielnames can ideally be different to the corrispondent field in the channel structure. Is bad practice but you're free to do it.

getFolderTree calls an update of the Block feature. just in case you changed the default file. 

updateParams now handles also the case when 'block', 'animal', 'tank' or 'sort' are passed as inputs.

Added updatePaths functions to tank and animal

StefanoBuccelli committed:
updatePaths feature

Now it is possible to update the whole folder tree in the structure hierarchy. This is useful when something changes or you have to move the tank folder to a new location.

defaults.Block added the field FileNames. Fielnames can ideally be different to the corrispondent field in the channel structure. Is bad practice but you're free to do it.

getFolderTree calls an update of the Block feature. just in case you changed the default file. 

updateParams now handles also the case when 'block', 'animal', 'tank' or 'sort' are passed as inputs.

Added updatePaths functions to tank and animal

Nabarb committed:
Added files for SPC

Nabarb committed:
small bugfixes

fixed some small bugs.

spikeimage - reverted back to simple progress circle (faster)
                     - future development: implement arrows movements

Nabarb committed:
Merge remote-tracking branch 'origin/feature/updatePaths' into feature/updatePaths

Nabarb committed:
Merge remote-tracking branch 'origin/feature/filtOptimisation' into feature/filtOptimisation

Nabarb committed:
fix typo

Nabarb committed:
Merge branch 'feature/updatePaths' into dev_IIT

m053m716 committed:
Begin "box parser" class that is specific to videos for Complex Box task etc.

m053m716 committed:
Merge pull request #6 from m053m716/feature/filtOptimisation

Feature/filt optimisation

Nabarb committed:
update files for SPC

update SPC to be more nigeLab oriented.
added find_temp function to return temperature parameters and temperature simulation results.

temperature_diag plots the different cluster sizes with respect to the temperature increase steps.

fixed run_cluster bugs regarding the binary lactation and saving location of logs and run-files.
Also code is a little better organized.

doautoClustering now prints out the progress %. And saves the results in a nigeLab compliant manner.

Nabarb committed:
Recluster update

Added recluster function to the Sorting interface.
Runs recluster on the selected class. Assigns noise to the noise cluster and all other newly discovered units to the first free slots in the clustering interface.

Nabarb committed:
bugfixes

FeatPopCallback now updates properly the y-axis features when the x-axis one is changed

FeaturesUi updates properly when a  new channel is selected

updateParams fixed a copypasta leftover

Nabarb committed:
SpikeImage navigator

SpikeImage is navigable with arrow keys now

Nabarb committed:
Toolboxes dependecies

Added a snippet of code to return the toolboxes  needed to run nigeLab

Nabarb committed:
bugfix

added SPCparams property in block

Nabarb committed:
Save & load updates

fixed a bug with the saving routine: blocks were saved three times, one time alone, another time inside the animal object and the third time inside the animal inside the tank. This is now fixed. 

Also everything is now save with v7 flag speeding up a lot the procedure.

Nabarb committed:
Merge pull request #7 from m053m716/feature/doAutoClustering

Feature/do auto clustering

Nabarb committed:
GUI first commit

Development of Dashboard started
    added panels with titles, nigelPanel
    Added the Dashboard
    renewed the clearSpace GUI
    added getopt for easy varargin parsing

Nabarb committed:
GUI first commit

Development of Dashboard started
    added panels with titles, nigelPanel
    Added the Dashboard
    renewed the clearSpace GUI
    added getopt for easy varargin parsing

Nabarb committed:
Bugfix DiskDataSave

When appending data using diskdata, the appending direction was ambiguous. This fixes it by appending data on the longest compatible direction, if not explicitly specified. This triggers a warning;
intan2Block Stim extraction is now working

Nabarb committed:
Added missing files

Added missing functions at different hierarchy level, needed for gui scalaility

Nabarb committed:
Added missing files

Added missing functions at different hierarchy level, needed for gui scalaility

Nabarb committed:
nigelColors

added default function to retrieve color throughout the  GUIs.
added nigelColor to the except parameters in updateParams

Nabarb committed:
nigelColors

added default function to retrieve color throughout the  GUIs.
added nigelColor to the except parameters in updateParams

Nabarb committed:
nigelColors contd

Nabarb committed:
nigelColors contd

Nabarb committed:
Remote Monitor

Added the monitor panel. The idea behind this is to spawn process in background (with the batch function) and have them write the state of the operation to a file. The file is read by the remote monitor and the progressbars are updated.
This works for process running in background on the local machine, I still have to figure out a way to get this working on a remote machine. 
The progressFiles (ie the files where the progress is stored) are created in the temp folder of the system. They are binary files with the first element being a int32 that encodes the total number of steps the operation is going to do. after that each time a step is done a uint8 1 is added to the files. By counting how many of those are added you get the progress till now. 

The progressMonitor relies on a timer to update the progressBars.

Nabarb committed:
Remote Monitor

Added the monitor panel. The idea behind this is to spawn process in background (with the batch function) and have them write the state of the operation to a file. The file is read by the remote monitor and the progressbars are updated.
This works for process running in background on the local machine, I still have to figure out a way to get this working on a remote machine. 
The progressFiles (ie the files where the progress is stored) are created in the temp folder of the system. They are binary files with the first element being a int32 that encodes the total number of steps the operation is going to do. after that each time a step is done a uint8 1 is added to the files. By counting how many of those are added you get the progress till now. 

The progressMonitor relies on a timer to update the progressBars.

Nabarb committed:
Silhuette

Better silhouette visualization. Now you can visualize the score for all the plotted dots (it was the cluster mean before).
Computing the silhouette score is still computational expensive. Might benefit from gpu support.

 (Distance menu selector is broken at this point in time)

Nabarb committed:
Silhuette

Better silhouette visualization. Now you can visualize the score for all the plotted dots (it was the cluster mean before).
Computing the silhouette score is still computational expensive. Might benefit from gpu support.

 (Distance menu selector is broken at this point in time)

Nabarb committed:
SNEO optimization 

Tentative sneo optimization. The most computational expensive step was addressed. At a first look it looks like a factor of 3 is gained in computation time (for this step).

Nabarb committed:
SNEO optimization 

Tentative sneo optimization. The most computational expensive step was addressed. At a first look it looks like a factor of 3 is gained in computation time (for this step).

Nabarb committed:
Queue monitor update

Tempdir                 default temp directory path
remoteMonitor    added progress monitor for remote works. At this point uses a local file to keep track of progress. I'd like to use communicating jobs in the future. 
nigelPanel + attachScrollPanel nigelpanels can now be made scrollable!

Nabarb committed:
Queue monitor update

Tempdir                 default temp directory path
remoteMonitor    added progress monitor for remote works. At this point uses a local file to keep track of progress. I'd like to use communicating jobs in the future. 
nigelPanel + attachScrollPanel nigelpanels can now be made scrollable!

Nabarb committed:
Added support for progress monitoring

All the doThings now are using files to track progress

Nabarb committed:
Added support for progress monitoring

All the doThings now are using files to track progress

Nabarb committed:
Sort Save bugfix

Nabarb committed:
Bugfixes

Updated filesToSkip yo updateParams
added progresstracking to intan2Block

Nabarb committed:
Bugfixes

Updated filesToSkip yo updateParams
added progresstracking to intan2Block

Nabarb committed:
plotQuality bugfix

Improved barplots in plotquality.

Nabarb committed:
plotQuality bugfix

Improved barplots in plotquality.

Nabarb committed:
ClearSpace bugfixes

Fixed some logical bugs in clearSpace

Nabarb committed:
ClearSpace bugfixes

Fixed some logical bugs in clearSpace

Nabarb committed:
DashBoard improvements

Added the titleBar. No callbacks yet.

Added the panelGroup for parameters visualization

Nabarb committed:
DashBoard improvements

Added the titleBar. No callbacks yet.

Added the panelGroup for parameters visualization

Nabarb committed:
HighDimsUI update

Added high dimensional visualization to featureUI. adapted from DataHIgh. Instead of plotting the features to a plane parallel to some axis, now there is the possibility to rotate such plane. Some Projections can, in fact, be better suited for the clustering process.

Nabarb committed:
HighDimsUI update

Added high dimensional visualization to featureUI. adapted from DataHIgh. Instead of plotting the features to a plane parallel to some axis, now there is the possibility to rotate such plane. Some Projections can, in fact, be better suited for the clustering process.

Nabarb committed:
plotQuality update

The bars corresponding to the clusters now disappear according to the visibility of said clusters

Nabarb committed:
plotQuality update

The bars corresponding to the clusters now disappear according to the visibility of said clusters

Nabarb committed:
Merge remote-tracking branch 'origin/feature/GUI' into feature/GUI

StefanoBuccelli committed:
bypassing toolboxes: imageprocessing

imageprocessing: imfreehand replaced by freehanddraw both in FeaturesUI and SpikeImage
statistic and machine learning: combnk replaced by nchoosek in Sort.initUI

StefanoBuccelli committed:
in DashBoard.m check that you have the licence for the parallel computing toolbox

instead of running batch you run feval.

Tempdir was also modified just make sure TempDir  is created.

Still open issue: The way you plot the progress bar in Queue panel

Co-Authored-By: nabarb <nabarb@users.noreply.github.com>

m053m716 committed:
Uncommitted files, but unsure when modifications are from so just committing on KUMC branch. -MM

StefanoBuccelli committed:
tempDir case-sensitive typo solved

StefanoBuccelli committed:
added demo files and modified block to be consistent with mytank demo

StefanoBuccelli committed:
Merge pull request #18 from m053m716/feature/GUI

Feature/gui

Nabarb committed:
Bugfixes

Recluster now works, now DoSPC takes care of the clustering procedure. defaults updated.

Fixed Dashboard compatibility issue.

Fixed some graphical issues in HighDimsUI

StefanoBuccelli committed:
bugfix for dashboard (FB) + gitIgnore demo files uzipped

StefanoBuccelli committed:
added Widget Toolbox in utils folder and modified intall_nigeLab.m to install it

Nabarb committed:
Clustering bugfixes

Fixed some bugs afflicting AutoClustering and code slightly cleaned.

Nabarb committed:
Shorter filenames

Now filenames are shorter. This prevent a stupid bug on windows where filenames have a maximum length, but breaks all backwards compatibility! Pull at your own risk!!

Nabarb committed:
Small demo enhancement

Now demo is completely automatized

StefanoBuccelli committed:
making signal processing toolbox mandatory, you get an error if licence is not present

StefanoBuccelli committed:
Merge branch 'dev_IIT' of https://github.com/m053m716/ePhys_packages into dev_IIT

m053m716 committed:
Change MenuSelectedFcn to Callback for compatibility with R2017a

m053m716 committed:
Merge remote-tracking branch 'origin/feature/ShortFileNames' into feature/qOperations

m053m716 committed:
Add 'UseParallel' flag to Queue parameters. This flag will be used to delineate between parallel and serial execution.
Add 'getChildPanel' method to DashBoard class. This is just to facilitate indexing by Tag. Changed the Tag strings for nigelPanel objects to make them less confusing.

m053m716 committed:
Add 'findGoodCluster' and 'jobFinishedAlert' to utils, as helper functions for parallel processing.
Add 'play' function to +sounds, to allow easy "notification sound" play based on filenames without having to reconstruct the path to the sound file each time.
Change default Queue parameters.
Begin modifications to "do" functions to accomodate progress bar updates. "do" functions need to individually set the job tag as they are updated, allowing the remoteJobMonitor to listen for Tag property changes. If the Tag is not updated dynamically as the "do" function is completed, then at the end, the jobFinishedFunction should trigger an update to the remoteJobMonitor to ensure that the status bar is completed in the "Queue" panel.

m053m716 committed:
Merge branch 'KUMC' into feature/qOperations

m053m716 committed:
Add getNumBlocks method to Tank, Animal, and Block classes to make it easy to count total number of blocks, no matter which class is passed.
Move tankGet and tankSet methods to deprecated folder.
Move initial attempt at "remoteMonitor" to "oldRemoteMonitor" and put in deprecated sub-folder. This attempt will use event listeners with the Tag property.

m053m716 committed:
Merge pull request #21 from m053m716/feature/ShortFileNames

Shorter filenames

m053m716 committed:
Modifications to try and register progress updates on remote workers.

Nabarb committed:
Bugfix

for some reason I added this check a month ago. If the data recording  is triggered from something the time vector has negative values in it. If this check active, nothing would be saved.

StefanoBuccelli committed:
adding compiled FilterX for MacOS High Sierra 10.13.6

to compile, the only way is to use

mex -compatibleArrayDims FilterX.c

Nabarb committed:
updatePaths updated to move files

When updatePaths is called now it moves automatically any file found in the canonical path (ie the one defined in the Paths structure)

Nabarb committed:
updatePaths updated to move files

When updatePaths is called now it moves automatically any file found in the canonical path (ie the one defined in the Paths structure)

Nabarb committed:
Merge pull request #23 from m053m716/feature/ShortFileNames

updatePaths updated to move files

StefanoBuccelli committed:
tankSet needs an additional argument (value) as already present in tankSet.m itself

Nabarb committed:
Animal name listener

Added listener on animal name. Updates all Meta properties of child blocks when changed

Nabarb committed:
Pars Panel first commit

Pars Panel is now able to visualize all parameters in the Pars structure inside tank,block or animal. 

I slightly changed Block in  order to add a Pars public property that contains all the other parameters fields. This is a "tappullo", but I think this design is better and more versatile than having all the parameters structure living in some ad hoc properties of the block. We should probably switch to this.

m053m716 committed:
Attempt to debug updateRemotMonitor method.

m053m716 committed:
Merge remote-tracking branch 'origin/feature/GuiParamsTab' into feature/qOperations

m053m716 committed:
Merge branch 'dev_IIT' into feature/qOperations

# Conflicts:
#	+nigeLab/@Tank/Tank.m
#	.gitignore

Commit:
Remove block folder that is at the wrong level for some reason?

Nabarb committed:
Params Panel bugfix

The root node in tree in the Overview panel was selceted before the Params panel was initialized. This triggered the update of a non yet existent uitabgroup. This fixes #25.

Nabarb committed:
Params Panel bugfix

The root node in tree in the Overview panel was selceted before the Params panel was initialized. This triggered the update of a non yet existent uitabgroup. This fixes #25.

Nabarb committed:
Merge remote-tracking branch 'origin/feature/GuiParamsTab' into feature/GuiParamsTab

m053m716 committed:
Change formatting, naming conventions for parameters of Queue.

m053m716 committed:
Merge remote-tracking branch 'origin/feature/GuiParamsTab' into feature/qOperations

m053m716 committed:
Merge remote-tracking branch 'origin/master' into feature/qOperations

# Conflicts:
#	+nigeLab/img/Nigel.png

m053m716 committed:
Works for doRawExtraction, but only with UseRemote set to false. Also there is some weird glitch with the progress bar going backwards in progress briefly that needs to be resolved.. but anyways the main thing is that it's reading progress from the workers now! -MM

m053m716 committed:
Having issues with doUnitFilter, which takes a very long time to run for some reason and won't complete? doRawExtraction works. -MM

m053m716 committed:
Fix bug where progress bar was reset inadvertently for doRawExtraction (intan2Block). -MM

m053m716 committed:
Add runFun command to run block method on all child blocks of a tank.
Add checkMask, mostly as a debug tool, as the Mask was not being set properly sometimes.

m053m716 committed:
Update doAutoClustering and doLFPExtraction to include the blockObj.notifyUser method.

StefanoBuccelli commented:
#31  @m053m716 I was trying to run doRawExtraction with this branch. I don't have Parallel Toolbox. 

> 'getCurrentJob' requires Parallel Computing Toolbox.

calling 'getCurrentJob' in nigeLab.Block/notifyUser (line 34). 

Called by: nigeLab.Block/intan2Block (line 140)

please make sure it runs without that toolbox :)

m053m716 committed:
Changed to check compatibility for getCurrentJob in notifyUser. Removed passing job as function argument (does not work in Matlab). -MM

m053m716 commented:
> 
> 
> #31 @m053m716 I was trying to run doRawExtraction with this branch. I don't have Parallel Toolbox.
> 
> > 'getCurrentJob' requires Parallel Computing Toolbox.
> 
> calling 'getCurrentJob' in nigeLab.Block/notifyUser (line 34).
> 
> Called by: nigeLab.Block/intan2Block (line 140)
> 
> please make sure it runs without that toolbox :)

Thanks for catching that. I just pushed a new commit that should handle that and check for the parallel license in notifyUser as well.

StefanoBuccelli committed:
added pars.UseParallel in defaults/notifications

StefanoBuccelli commented:
> > #31 @m053m716 I was trying to run doRawExtraction with this branch. I don't have Parallel Toolbox.
> > > 'getCurrentJob' requires Parallel Computing Toolbox.
> > 
> > 
> > calling 'getCurrentJob' in nigeLab.Block/notifyUser (line 34).
> > Called by: nigeLab.Block/intan2Block (line 140)
> > please make sure it runs without that toolbox :)
> 
> Thanks for catching that. I just pushed a new commit that should handle that and check for the parallel license in notifyUser as well.

Thanks!
I added pars.UseParallel =0 as defaults/notification.. does it make sense? @m053m716

StefanoBuccelli commented:
> > > #31 @m053m716 I was trying to run doRawExtraction with this branch. I don't have Parallel Toolbox.
> > > > 'getCurrentJob' requires Parallel Computing Toolbox.
> > > 
> > > 
> > > calling 'getCurrentJob' in nigeLab.Block/notifyUser (line 34).
> > > Called by: nigeLab.Block/intan2Block (line 140)
> > > please make sure it runs without that toolbox :)
> > 
> > 
> > Thanks for catching that. I just pushed a new commit that should handle that and check for the parallel license in notifyUser as well.
> 
> Thanks!
> I added pars.UseParallel =0 as defaults/notification.. does it make sense? @m053m716

I tried both 0 and 1 but in both cases it does not extract raw files correctly.. always extracting 000%

m053m716 commented:
Is it just the progress bar that fails to update or do the files get created, it just fails to update the status?

StefanoBuccelli commented:
> Is it just the progress bar that fails to update or do the files get created, it just fails to update the status?

Oh, you're right. just the progress bar.. great!

m053m716 committed:
Update recent changes. -MM

Nabarb committed:
UNC update

All path are in UNC convention now. No idea what will happen on linux machines. Testing needed.

Nabarb committed:
UNC update 2

Now all the paths follow the UNC convention.

Block,Animal and Tank use UNC alse for RecFile (or equivalent).

genPaths and getFolderTree don't need the UseRemote flag anymore, removed

Nabarb committed:
reportProgress

reportProgress takes the place of notifyUser (sorry, my bad).

Updated default/Notifications to reflect the new notification system

added a second output argument in jobTag2Pct

Nabarb committed:
Remote worker update

Remote workers now work if they have access to the code. This is achieved by programmatically creating an addpath script with the path (UNC) to the folder containing the code. 

Every doMethod should contain the configuration snippet to work correctly.

Nabarb committed:
minor update

intan2Block updated t use the new notification system.

typo in linkFilt. (Are these even still in use? #32)

Nabarb committed:
remoteMonitor update

remoteMonitor moved to his own class.
Now a functional remoteMonitor can be created in any figure, panel or nigelPanel. 

Bars can be added with the addBar(string,job) method. lacking documentation, an update will follow.

Added undocumentedMatlab findjObj to the utils. This tools is amazing and  returns the jObj associated with matlab graphical obj. useful for customization.

Nabarb committed:
Fixed graphical glitch in dashoard + remoteMonitor

Nabarb committed:
jobCompleted event update

The completion of a job triggers the barCompleted function of remoteMonitor. This riggers a job competed which is caught by the dashboard listener to whatever it needs to do, at this point only refreshing the corresponding block to get the correct Status. 

Also some cosmetic adjustments.

Nabarb committed:
Fixed a bug preventing serial execution from dashboard

Dashboard now checks if the paralleled toolbox is installed (was only checking if it was licensed).

Nabarb committed:
Fixed a bug preventing serial execution from dashboard

Dashboard now checks if the paralleled toolbox is installed (was only checking if it was licensed). #8

Nabarb committed:
Dashboard serial update

When operations re executed serially from the dashbar, now bars are spawned in the Queue panel notifying the user to check the command window fr serial operations.

RemoteMonitor now has minimal functionalities for serial execution notification.

Nabarb committed:
Dashboard serial update

When operations re executed serially from the dashbar, now bars are spawned in the Queue panel notifying the user to check the command window fr serial operations.

RemoteMonitor now has minimal functionalities for serial execution notification.

Nabarb committed:
updated DashBoard serial execution

Updated the serial execution behavior to correctly reflect the RemoteMonitor class usage.

Nabarb committed:
updated DashBoard serial execution

Updated the serial execution behavior to correctly reflect the RemoteMonitor class usage.

Nabarb committed:
reportProgress Serial update

reportProgress can now be used both for serial and parallel operations. It will automatically detect the environment it's working in and switch between updating the text on the screen or printing it again.

Nabarb committed:
reportProgress Serial update

reportProgress can now be used both for serial and parallel operations. It will automatically detect the environment it's working in and switch between updating the text on the screen or printing it again.

Nabarb committed:
Merge remote-tracking branch 'origin/feature/qOperations' into feature/qOperations

Nabarb commented:
I'm going to close this and open another one based on the last workflow we agreed on @m053m716 .
